osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -240,6 +237,18 @@ private string getBPMRange(Beatmap beatmap)
                 return Math.Round(bpmMin) + "-" + Math.Round(bpmMax) + "bpm (mostly " + Math.Round(beatmap.ControlPointInfo.BPMMode) + "bpm)";
             }
 
+            private string getLength(Beatmap beatmap)
+            {
+                if (beatmap.HitObjects.Count == 0)
+                    return "-";
+
+                HitObject lastObject = beatmap.HitObjects.LastOrDefault();
+                double endTime = (lastObject as IHasEndTime)?.EndTime ?? lastObject?.StartTime ?? 0;
+                double length = Math.Min(endTime - beatmap.HitObjects.First().StartTime, TimeSpan.MaxValue.TotalMilliseconds - 1);

```

Comment:
we shouldn't be hitting infinite time in the first place. i believe the underlying problem should be fixed rather than making hack fixes like this.